packages/nuxt/src/pages/module.ts

```diff
@@ -186,6 +190,32 @@ export default defineNuxtModule({
       }
     })
 
+    const routeRulesMatcher = toRouteMatcher(
+      createRadixRouter({ routes: { ...nuxt.options.routeRules, ...nuxt.options.nitro?.routeRules } })
+    )
+    const getRouteRules = (path: string) =>
+      defu({}, ...routeRulesMatcher.matchAll(path).reverse()) as NitroRouteRules
+
+    nuxt.hook('build:manifest', async (manifest) => {
+      const pages = await resolvePagesRoutes()
+      await nuxt.callHook('pages:extend', pages)
+      const routes: string[] = []
+
+      for (const page of pages) {
+        const rules = getRouteRules(page.path)
+        if (rules.noScripts && page.file && !pages.some(p => p.file === page.file && p !== page)) {
+          routes.push(relative(nuxt.options.srcDir, page.file))
+        }
+      }
+
+      for (const file of routes) {
+        if (routes.some(fileName => manifest[fileName] && manifest[file].resourceType === 'script')) {
```

Comment:
I think we can simplify to:
```suggestion
        if (manifest[file].resourceType !== 'script') { continue }
        if (routes.some(fileName => fileName in manifest)) {
```